lib/src/rules/prefer_initializing_formals.dart

```diff
@@ -112,9 +119,30 @@ class _Visitor extends SimpleAstVisitor {
       return !(constructorFieldInitializer.fieldName.bestElement?.isPrivate ??
               true) &&
           expression is SimpleIdentifier &&
-          parameters.contains(expression.bestElement);
+          parameters.contains(expression.bestElement) &&
+          (!parametersUsedMoreThanOne.contains(expression.bestElement) ||
+              constructorFieldInitializer.fieldName.bestElement?.name ==
+                  expression.bestElement.name);
     }
 
+    void processElement(Element element) {
+      if (parametersUsedOne.contains(element)) {
```

Comment:
With sets, you can do this more efficiently by adding the element to the set and then using the value returned by `add` to determine whether it was already in the set.